src/main/scala/io/citrine/lolo/stats/StatsUtils.scala

```diff
@@ -0,0 +1,70 @@
+package io.citrine.lolo.stats
+
+import io.citrine.lolo.linear.LinearRegressionLearner
+
+import scala.util.Random
+
+object StatsUtils {
+
+  /** Compute the mean of a (weighted) vector, X */
+  def mean(X: Seq[Double], weights: Option[Seq[Double]] = None): Double = {
+    val actualWeights = weights.getOrElse(Seq.fill(X.length)(1.0))
```

Comment:
When there are no weights, how about returning the unweighted mean instead of allocating another array and doing more complicated arithmetic when we don't have to? I'm open to this being dismissed as an unnecessary performance optimization, but it's also just a one liner here and in the immediately subsequent methods.

Reply:
Is it just a one-liner? I thought it would involve an ugly if-else block for each method, which is why I decided to take the performance hit in exchange for simpler-looking methods.